atc/exec/set_pipeline_step_test.go

```diff
@@ -231,82 +233,94 @@ jobs:
-						fakePipeline.ConfigReturns(pipelineObject, nil)
+						fakeTeam.PipelineReturns(fakePipeline, true, nil)
 					})
 
-					It("should log no-diff", func() {
-						Expect(stdout).To(gbytes.Say("no diff found."))
+					Context("when no diff", func() {
+						BeforeEach(func() {
+							fakePipeline.ConfigReturns(pipelineObject, nil)
+						})
+
+						It("should log no-diff", func() {
+							Expect(stdout).To(gbytes.Say("no diff found."))
+						})
 					})
-				})
 
-				Context("when there are some diff", func() {
-					BeforeEach(func() {
-						pipelineObject.Jobs[0].PlanSequence[0].TaskConfig.Run.Args = []string{"hello world"}
-						fakePipeline.ConfigReturns(pipelineObject, nil)
+					Context("when there are some diff", func() {
+						BeforeEach(func() {
+							pipelineObject.Jobs[0].PlanSequence[0].TaskConfig.Run.Args = []string{"hello world"}
+							fakePipeline.ConfigReturns(pipelineObject, nil)
+						})
+
+						It("should log diff", func() {
+							Expect(stdout).To(gbytes.Say("job some-job has changed:"))
+						})
 					})
 
-					It("should log diff", func() {
-						Expect(stdout).To(gbytes.Say("job some-job has changed:"))
+					Context("when SavePipeline fails", func() {
+						BeforeEach(func() {
+							fakeTeam.SavePipelineReturns(nil, false, errors.New("failed to save"))
+						})
+
+						It("should return error", func() {
+							Expect(stepErr).To(HaveOccurred())
+							Expect(stepErr.Error()).To(Equal("failed to save"))
+						})
 					})
-				})
 
-				Context("when SavePipeline fails", func() {
-					BeforeEach(func() {
-						fakeTeam.SavePipelineReturns(nil, false, errors.New("failed to save"))
+					It("should save the pipeline un-paused", func() {
+						Expect(fakeTeam.SavePipelineCallCount()).To(Equal(1))
+						pipelineMetadata, _, _, paused := fakeTeam.SavePipelineArgsForCall(0)
+						Expect(pipelineMetadata.Name).To(Equal("some-pipeline"))
+						Expect(paused).To(BeFalse())
 					})
 
-					It("should return error", func() {
-						Expect(stepErr).To(HaveOccurred())
-						Expect(stepErr.Error()).To(Equal("failed to save"))
+					It("should stdout have message", func() {
+						Expect(stdout).To(gbytes.Say("setting pipeline: some-pipeline"))
+						Expect(stdout).To(gbytes.Say("done"))
 					})
-				})
 
-				It("should save the pipeline un-paused", func() {
-					Expect(fakeTeam.SavePipelineCallCount()).To(Equal(1))
-					name, _, _, paused := fakeTeam.SavePipelineArgsForCall(0)
-					Expect(name).To(Equal("some-pipeline"))
-					Expect(paused).To(BeFalse())
+					It("should finish successfully", func() {
+						Expect(fakeDelegate.FinishedCallCount()).To(Equal(1))
+						_, succeeded := fakeDelegate.FinishedArgsForCall(0)
+						Expect(succeeded).To(BeTrue())
+					})
 				})
 
-				It("should stdout have message", func() {
-					Expect(stdout).To(gbytes.Say("setting pipeline: some-pipeline"))
-					Expect(stdout).To(gbytes.Say("done"))
+				It("always saves Job and Build ID's", func() {
```

Comment:
Did some acceptance on this, and noticed it wasn't updating the Job and Build IDs when there is no diff - I think you'll need a few test cases here to make sure it gets called when:
1. There's no pipeline yet (what this is testing now, though not explicitly)
2. There is a pipeline and the config differs
3. There is a pipeline and the config is the same